stackrox/stackrox
# ROX-24330: Factor out reusable view components for compliance report

## Description

### Problem

Review step of scan config wizard and view page for scan config have unneeded differences in presentation of information.

### Analysis

Minimal differences for review step versus view page:

1. Review step does not need to repeat **Operator status** from the **Select clusters** step.

    * Although status seems relevant for **Select clusters** step, it might distract from confirming correct selection in **Review** step.

    * Although it is questionable whether status is relevant for compliance analyst persona on view page, keep it until **Platform Configuration** has a **Compliance Operator** page for a link from **System Health** page for platform administrator persona.

2. Review step does not need **Last run** nor **Last updated** pseudo parameters.

### Solution

1. Replace separate cards with one card.
2. Add `headingLevel` prop for review step versus view page.
3. Distinguish `lastUpdatedTime` from `lastExecutedTime` added in #11222

### Changes

1. Edit ScanConfigClustersTable.tsx file.

2. Rename and edit ScanConfigParametersView.tsx file.
    * Refactor props for form review versus view page.

3. Rename and edit ScanConfigProfilesView.tsx file.
    * Render count as `Badge` element.

4. Edit ScanConfigsTablePage.tsx file.
    * Add `dataLabel` props to `Td` elements.
    * Replace `lastUpdatedTime` with `lastExecutedTime` for **Last run** cell.

5. Edit Wizard/ReviewConfig.tsx file.
    * Render `ScanConfigParametersView` and `ScanConfigProfilesView` elements.
    * Replace `profiles` and `selectedProfiles` with `formikValues.profiles` because profiles do not have `id` versus `name` distinction like clusters.
    * For clusters, render count as `Badge` element and render cluster names inline in `List` element, because table with status is not relevant for review.
    * Add conditional rendering for `NotifierConfigurationView` element.

6. Edit ScanConfigWizardForm.tsx file.
    * Delete `profiles` prop from `ReviewConfig` element.

7. Edit ViewScanConfigDetail.tsx file.
    * Render **Last run** nor **Last updated** pseudo parameters as optional children of `ScanConfigParametersView` element.
    * Add conditional rendering for `NotifierConfigurationView` element.

8. Edit ComplianceScanConfigurationService.ts file.
    * Add `lastExecutedTime` property.

### Residue

1. Investigate absence of `notifiers` in response of GET /v2/compliance/scan/configurations request.
2. Investigate remaining differences in presentation of time information:
    * `ViewScanConfigDetail` component renders `Timestamp` elements.
    * `ScanConfigsTablePage` component renders result of `format(lastExecutedTime, 'DD MMM YYYY, h:mm:ss A')` function call.
3. Decide whether ScanConfigsTablePage.tsx file needs its own Table folder.

## Checklist

## Testing Performed

Prerequisite for `yarn start` in ui folder:

```sh
export ROX_COMPLIANCE_REPORTING=true
```

1. `yarn lint` in ui
2. `yarn build` in ui and then compute branch - master for the following
    * `wc build/static/js/*.js`
        main.js 0 = 4635494 - 4635494
        total -5223 = 11702099 - 11707322
    * `ls -al build/static/js/*.js | wc`
        files 0 = 172 - 172
3. `yarn start` in ui

### Manual testing

1. Visit /main/compliance/schedules and create a scan schedule.
    * See GET /v2/compliance/integrations request for step 1.
    * See GET /v1/notifiers request for step 4.

2. Add a delivery destination in **Configure report** step.
    ![Configure_report](https://github.com/stackrox/stackrox/assets/11862657/f2d16ce2-1fde-4653-9323-ecd1cc9f87e3)

3. Click **Next** to see **Review and create** step.

    * More differences before changes:
        ![ReviewConfig_separate](https://github.com/stackrox/stackrox/assets/11862657/6a7fa911-2541-4715-8ff1-98902f8d4b76)

    * Less difference after changes:
        ![ReviewConfig_unified](https://github.com/stackrox/stackrox/assets/11862657/049534c6-cfe9-473d-9b68-4c42ee04a06b)

4. Click **Create** and see compliance scan schedules table.
    ![ViewScanConfigsTablePage](https://github.com/stackrox/stackrox/assets/11862657/4fcec786-7567-40a9-b1cb-4777af3e951f)

5. Click link to see view page.

    * More differences before changes:
        ![ViewScanConfigDetail_separate](https://github.com/stackrox/stackrox/assets/11862657/46780729-61ab-4730-8f81-038609c486f3)

    * Less difference after changes, but **No delivery destinations**
        ![ViewScanConfigDetail_unified](https://github.com/stackrox/stackrox/assets/11862657/9243ff4d-3667-4546-8fbc-1076c206cb9a)
